pkgs/applications/networking/jdownloader2/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv, lib, fetchurl, writeScript, makeDesktopItem, copyDesktopItems, jre, imagemagick, coreutils }:
+
+let
+
+  icon = fetchurl {
+    url = "https://jdownloader.org/_media/vote/trazo.png";
+    sha256 = "3ebab992e7dd04ffcb6c30fee1a7e2b43f3537cb2b22124b30325d25bffdac29";
+  };
+
+  wrapper = writeScript "jdownloader" ''
+    #! ${stdenv.shell}
+    PATH=${lib.makeBinPath [ jre coreutils ]}
+    JDJAR=''${XDG_DATA_HOME:-$HOME/.local/share}/jdownloader/JDownloader.jar
+    dir=`dirname "$0"`
+    if [ ! -f ''${JDJAR} ]; then
+        install -Dm755 $dir/JDownloader.jar ''${JDJAR}
+    fi
+    ${jre}/bin/java -jar ''${JDJAR} "''${@}"
+  '';
+
+in stdenv.mkDerivation rec {
+  pname = "jdownloader2";
+  version = "2.0";
+
+  #there is no https endpoint for this domain
+  src = fetchurl {
+    url = "https://archive.org/download/jdownloader_202109/JDownloader.jar";
+    sha256 = "9951b786e24fc3777a0df0a7b516ba53d0c8e778d6a69ebc29dcff86ee6b5829";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ imagemagick copyDesktopItems ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "JDownloader 2";
+      exec = wrapper;
+      icon = "jdownloader";
+      comment = "Free, open-source download management tool.";
```

Comment:
imho, "FOSS" is not relevant here. i would just say "Download manager"
(you can download all kinds of files with jd, also youtube videos ...)

Reply:
I took that from the official website, the FOSS refers to JDownloader not the stuff you can download, I think. Sure thing, I can change that.

Reply:
As for the package name, I can change that as well. For quite some time there were jdownloader and jdownloader2, the jdownloader is obsolete by now, but few people still use the jdownloader2 name as you can see on the repology https://repology.org/projects/?search=jdownloader